diff --git a/init.lua b/init.lua
index 0fda20c..897ddae 100644
--- a/init.lua
+++ b/init.lua
@@ -1,5 +1,5 @@
 vim.g.base46_cache = vim.fn.stdpath "data" .. "/base46/"
-vim.g.mapleader = " "
+vim.g.mapleader = ","
 
 -- bootstrap lazy and all plugins
 local lazypath = vim.fn.stdpath "data" .. "/lazy/lazy.nvim"
diff --git a/lua/chadrc.lua b/lua/chadrc.lua
index c82dca1..edabe2a 100644
--- a/lua/chadrc.lua
+++ b/lua/chadrc.lua
@@ -6,7 +6,7 @@
 local M = {}
 
 M.base46 = {
-	theme = "onedark",
+	theme = "everblush",
 
 	-- hl_override = {
 	-- 	Comment = { italic = true },
@@ -14,4 +14,12 @@ M.base46 = {
 	-- },
 }
 
+M.term = {
+   float = {
+     row = 0, col = 0,
+     width = 1, height = 0.9,
+     border = "single",
+   },
+}
+
 return M
diff --git a/lua/configs/lspconfig.lua b/lua/configs/lspconfig.lua
index 478df01..e2d2bb1 100644
--- a/lua/configs/lspconfig.lua
+++ b/lua/configs/lspconfig.lua
@@ -4,7 +4,7 @@ require("nvchad.configs.lspconfig").defaults()
 local lspconfig = require "lspconfig"
 
 -- EXAMPLE
-local servers = { "html", "cssls" }
+local servers = { "html", "cssls", "vimls", "lua_ls", "marksman", "bashls", "clangd", "pyright", "gopls", "ts_ls", "rust_analyzer" }
 local nvlsp = require "nvchad.configs.lspconfig"
 
 -- lsps with default config
diff --git a/lua/mappings.lua b/lua/mappings.lua
index 783b78f..8b86adb 100644
--- a/lua/mappings.lua
+++ b/lua/mappings.lua
@@ -4,7 +4,20 @@ require "nvchad.mappings"
 
 local map = vim.keymap.set
 
-map("n", ";", ":", { desc = "CMD enter command mode" })
+-- map("n", ";", ":", { desc = "CMD enter command mode" })
 map("i", "jk", "<ESC>")
 
+map("t", "<Esc>", "<C-\\><C-n>", {desc = "Exit terminal mode"})
+
+-- plugin keymap
+map("n", "<leader>n", "<cmd> NvimTreeToggle <cr>")
+
+local builtin = require('telescope.builtin')
+map("n", "<leader>fc", builtin.command_history, { desc = 'Telescope commands history' })
+map("n", "<leader>fs", builtin.lsp_document_symbols, { desc = 'Lists LSP document symbols in the current buffer' })
+
 -- map({ "n", "i", "v" }, "<C-s>", "<cmd> w <cr>")
+-- Disable mappings
+local nomap = vim.keymap.del
+
+nomap("n", "<C-n>")
diff --git a/lua/options.lua b/lua/options.lua
index 738f20b..38b74f2 100644
--- a/lua/options.lua
+++ b/lua/options.lua
@@ -2,5 +2,6 @@ require "nvchad.options"
 
 -- add yours here!
 
--- local o = vim.o
+local o = vim.o
+o.wrap = false
 -- o.cursorlineopt ='both' -- to enable cursorline!
diff --git a/lua/plugins/init.lua b/lua/plugins/init.lua
index 6474776..7719445 100644
--- a/lua/plugins/init.lua
+++ b/lua/plugins/init.lua
@@ -1,9 +1,8 @@
-return {
-  {
-    "stevearc/conform.nvim",
-    -- event = 'BufWritePre', -- uncomment for format on save
-    opts = require "configs.conform",
-  },
+return { {
+  "stevearc/conform.nvim",
+  -- event = 'BufWritePre', -- uncomment for format on save
+  opts = require "configs.conform",
+},
 
   -- These are some examples, uncomment them if you want to see them work!
   {
@@ -13,13 +12,55 @@ return {
     end,
   },
 
-  -- {
-  -- 	"nvim-treesitter/nvim-treesitter",
-  -- 	opts = {
-  -- 		ensure_installed = {
-  -- 			"vim", "lua", "vimdoc",
-  --      "html", "css"
-  -- 		},
-  -- 	},
-  -- },
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = {
+      ensure_installed = {
+        "vim", "lua", "vimdoc",
+        "html", "css", "c", "cpp",
+        "python", "markdown", "bash"
+      },
+    },
+  },
+
+  {
+    "easymotion/vim-easymotion",
+    config = function()
+      vim.g.EasyMotion_do_mapping = 0
+    end,
+    lazy = false,
+    keys = {
+      { "s",         "<Plug>(easymotion-overwin-f2)", mode = "n" },
+      { "<leader>j", "<Plug>(easymotion-j)",          mode = "n" },
+      { "<leader>k", "<Plug>(easymotion-k)",          mode = "n" },
+    },
+  },
+
+  {
+    "mg979/vim-visual-multi",
+    lazy = false,
+  },
+
+  {
+    "lervag/vimtex",
+    lazy = false, -- we don't want to lazy load VimTeX
+    -- tag = "v2.15", -- uncomment to pin to a specific release
+    init = function()
+      -- VimTeX configuration goes here, e.g.
+      vim.g.vimtex_view_method = "skim"
+      vim.g.vimtex_view_skim_sync = 1
+      vim.g.vimtex_view_skim_activate = 1
+      vim.g.vimtex_quickfix_mode = 0
+    end
+  },
+
+  {
+    "liuchengxu/vista.vim",
+    config = function ()
+      vim.g.vista_default_executive = "nvim_lsp"
+      vim.api.nvim_set_keymap("n", "<leader>o", ":Vista!!<CR>", { noremap = true, silent = true })
+    end,
+    lazy = false,
+  }
+
 }
